### PR TITLE
Minimal CR for resiliency

### PR DESCRIPTION
**Pull request by adarsh-dell**
# Description
Added support for the `resiliency` module in the minimal CR for the `PowerStore` driver. 
Minimal CR allows users to simply enable or disable the module, with only an enables field under each module’s spec—no components, arguments, or environment variables are needed. Since no changes were required in the core logic and the code was already designed to handle this case, the only requirement was to add default values that will be prepopulated and can be overwritten by the values from the CR.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
|https://github.com/dell/csm/issues/1449|

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Deployed the CSM 1.11.0 drivers (v2.11.0 CSI-PowerStore) and ensured that the driver starts up properly this test backward compatibility.
- [x] Updated the value in the old CR (v2.11.0) to verify that the default value is being overwritten, giving preference to the CR value over the prepopulated module config.
- [x] Installed the driver with the minimal configuration of the resiliency module and conducted basic sanity testing. This included taking down the interface where the pod was scheduled, monitored by podmon, and checking the logs to ensure the connectivity status is false. Taints were added to the node, the pod was rescheduled to another node, and once the node’s interface came back up, the taints were removed and the same connectivity status was reflected in the logs.

image (40)

image (39)

image (38)

image (37)

### CODE REVIEW

**Comment by adarsh-dell** [comment]
E2E is being covered under https://github.com/dell/csm-operator/pull/690/files.
Thanks

**Review by suryagupta4 — DISMISSED** [review]

**Review by meggm — APPROVED** [review]

File: operatorconfig/moduleconfig/resiliency/v1.11.0/container-powerstore-controller.yaml [hunk 1]
```diff
@@ -14,8 +14,20 @@
...
+image: dellemc/podmon:v1.11.0
 imagePullPolicy: IfNotPresent
+args:
+  - "--labelvalue=csi-powerstore"
+  - "--arrayConnectivityPollRate=60"
+  - "--skipArrayConnectionValidation=false"
+  - "--driverPodLabelValue=dell-storage"
+  - "--ignoreVolumelessPods=false"
+  - "--arrayConnectivityConnectionLossThreshold=3"
+  # Below 4 args should not be modified.
```
Comment (AkshaySainiDell) [inline operatorconfig/moduleconfig/resiliency/v1.11.0/container-powerstore-controller.yaml:26]: This comment is not required in the operatorconfig.

File: operatorconfig/moduleconfig/resiliency/v1.11.0/container-powerstore-node.yaml [hunk 2]
```diff
@@ -19,9 +19,25 @@ securityContext:
...
-image: dellemc/podmon:v1.10.0
+image: dellemc/podmon:v1.11.0
 imagePullPolicy: IfNotPresent
+args:
+  - "--labelvalue=csi-powerstore"
+  - "--arrayConnectivityPollRate=60"
+  - "--leaderelection=false"
+  - "--driverPodLabelValue=dell-storage"
+  - "--ignoreVolumelessPods=false"
+  # Below 4 args should not be modified.
```
Comment (AkshaySainiDell) [inline operatorconfig/moduleconfig/resiliency/v1.11.0/container-powerstore-node.yaml:30]: Same as above, this comment is not required in the operatorconfig.

File: operatorconfig/moduleconfig/resiliency/v1.11.0/container-powerstore-node.yaml [hunk 3]
```diff
@@ -19,9 +19,25 @@ securityContext:
...
+  - "--csisock=unix:/var/lib/kubelet/plugins/csi-powerstore.dellemc.com/csi_sock"
+  - "--mode=node"
+  - "--driver-config-params=/powerstore-config-params/driver-config-params.yaml"
+  - "--driverPath=csi-powerstore.dellemc.com"
 env:
+  # podmonAPIPort: Defines the port to be used within the kubernetes cluster
+  # Allowed values: Any valid and free port (string)
+  # Default value: 8083
```
Comment (AkshaySainiDell) [inline operatorconfig/moduleconfig/resiliency/v1.11.0/container-powerstore-node.yaml:38]: These comment are not required in the operatorconfig.

File: pkg/modules/resiliency.go [hunk 4]
```diff
@@ -37,6 +37,11 @@ var (
 	XCSIPodmonEnabled = "X_CSI_PODMON_ENABLED"
 )
 
+const (
+	ControllerMode = "controller"
+	NodeMode       = "node"
```
Comment (AkshaySainiDell) [inline pkg/modules/resiliency.go:42]: It seems these constants are utilized within this package. Please check if we can make them private.

**Review by AkshaySainiDell — CHANGES_REQUESTED** [review]

**Review by AkshaySainiDell — APPROVED** [review]

**Review by ashleyvjoy — APPROVED** [review]

**Review by suryagupta4 — APPROVED** [review]

**Review by chimanjain — APPROVED** [review]

**Pull request merged**